### ISSUE TITLE
fqcn[deep] - checks for directory pattern in tests and reports false positive

### ISSUE THREAD

**Issue by KB-perByte** [ansible/ansible-lint#4160]
##### Summary

It checks for directory patterns in tests and reports false positive warning about a deep plugin directory under the tests/ dir
The test directory should not be considered for this rule. 

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

N/A

```console (paste below)

```

##### Desired Behavior

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

![image](https://github.com/ansible/ansible-lint/assets/19584197/39245d29-8e35-4514-ad26-cef76e7f305a)

Please give some details of what is happening. Include a [minimum complete
verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by audgirka** [comment]
Related: 
https://github.com/ansible/ansible-lint/issues/2584
https://github.com/ansible/ansible-lint/pull/3502

**Comment by felixfontein** [comment]
Another false positive (https://dev.azure.com/ansible/ansible.posix/_build/results?buildId=114105&view=logs&j=2671e6a6-f41a-533c-2720-8ffdcf8ab96f&t=5b604a49-baaa-558f-6ab2-4a2ff646af4f) are files in `tests/unit/plugins/action/fixtures/`.

I wonder why this rule checks **unit tests** at all, since unit tests are known to come with fixtures that are in subdirectories.
